### ISSUE TITLE
[BUG] "Invalid QR code" "Make sure you're scanning a QR code displayed WhatsApp Web, ..." 

### ISSUE THREAD

**Issue by HUGEIT** [adiwajshing/Baileys#943]
**Describe the bug**
the QR scan can fail with Baileys. **I have finally reproduced this with "trace"** - I post the log at the bottom. clean clone, nothing else involved

this is **critical** because every time this happens, a session is created that you cannot find under "linked devices" and therefore cannot delete this. in Baileys this session can also be somewhat active, you still get notifications for `contacts.update` for example if you kill and run the process but there is no history sync or message sync.

you can only remove the session if you use a previous build of whatsapp where you can still click "Leave Beta". but these previous builds may be gone tomorrow so **I would love to help to address as long as we can before we burn through phone numbers for each attempt.**

**To Reproduce**
`1.` clone the project
`2.` `npm i`, `npm run example`
`3.` Scan the QR code, in my case with an iPhone

**Expected behavior**
the login should work every time like it does with the old Baileys

**Environment (please complete the following information):**
 - Is this on a server? no
 - What do your `connectOptions` look like? example
 - Do you have multiple clients on the same IP? no
 - Are you using a proxy? no

**Additional context**
I have only experienced this on iPhone and I am unable to reproduce the same behavior on whatsapp web.

I will post the full log below but basically it looks like the server sends something that Baileys doesn't acknowledge with another message or if there is a message that the server expects but Baileys forgets to send.

@adiwajshing what other information can you need regarding this?

```
baileysNov28 hu$ npm run example

> @adiwajshing/baileys-md@4.0.0 example
> node --inspect -r ts-node/register Example/example.ts

Debugger listening on ws://127.0.0.1:9229/545cc52d-f928-403e-bc74-9a945d9d0201
For help, see: https://nodejs.org/en/docs/inspector
connection update {
  connection: 'connecting',
  receivedPendingNotifications: false,
  qr: undefined
}
{"level":30,"time":1638202390291,"pid":88726,"hostname":"host","msg":"connected to WA Web"}
{"level":20,"time":1638202390406,"pid":88726,"hostname":"host","msg":"handshake recv from WA Web"}
{"level":30,"time":1638202390414,"pid":88726,"hostname":"host","msg":"handshake complete"}
{"level":30,"time":1638202390414,"pid":88726,"hostname":"host","msg":"not logged in, attempting registration..."}
{"level":10,"time":1638202390817,"pid":88726,"hostname":"host","msgId":"3134682111","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"set","id":"3134682111","xmlns":"md"},"content":[{"tag":"pair-device","attrs":{},"content":[{"tag":"ref","attrs":{},"content":{"type":"Buffer","data":[___________]}}]}]},"msg":"communication"}
connection update {
  qr: '___________'
}

{"level":10,"time":1638202395013,"pid":88726,"hostname":"host","msgId":"1788470798","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"set","id":"1788470798","xmlns":"md"},"content":[{"tag":"pair-success","attrs":{},"content":[{"tag":"platform","attrs":{"name":"iphone"},"content":null},{"tag":"device-identity","attrs":{},"content":{"type":"Buffer","data":[___________]}},{"tag":"device","attrs":{"jid":"__number__:42@s.whatsapp.net"},"content":null}]}]},"msg":"communication"}
{"level":20,"time":1638202395013,"pid":88726,"hostname":"host","msg":"pair success recv"}
{"level":20,"time":1638202395039,"pid":88726,"hostname":"host","msg":"pairing configured successfully"}
{"level":30,"time":1638202395039,"pid":88726,"hostname":"host","jid":"__number__:42@s.whatsapp.net","msg":"registered connection, restart server"}
saving auth state
connection update { isNewLogin: true, qr: undefined }
{"level":30,"time":1638202395041,"pid":88726,"hostname":"host","error":{"data":null,"isBoom":true,"isServer":false,"output":{"statusCode":410,"payload":{"statusCode":410,"error":"Gone","message":"Restart Required"},"headers":{}}},"msg":"connection closed"}
connection update {
  connection: 'close',
  lastDisconnect: {
    error: Error: Restart Required
        at WebSocket.<anonymous> (baileysNov28/src/Socket/socket.ts:491:17) {
      data: null,
      isBoom: true,
      isServer: false,
      output: [Object]
    },
    date: 2021-11-29T16:13:15.042Z
  }
}
connection update {
  connection: 'connecting',
  receivedPendingNotifications: false,
  qr: undefined
}
{"level":30,"time":1638202395175,"pid":88726,"hostname":"host","msg":"connected to WA Web"}
{"level":20,"time":1638202395212,"pid":88726,"hostname":"host","msg":"handshake recv from WA Web"}
{"level":30,"time":1638202395232,"pid":88726,"hostname":"host","msg":"handshake complete"}
{"level":30,"time":1638202395232,"pid":88726,"hostname":"host","msg":"logging in..."}
{"level":10,"time":1638202395603,"pid":88726,"hostname":"host","fromMe":false,"frame":{"tag":"success","attrs":{"t":"1638202395","props":"26","location":"odn","abprops":"0","creation":"1638202394","companion_enc_static":"___________"},"content":null},"msg":"communication"}
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
{"level":30,"time":1638202395655,"pid":88726,"hostname":"host","msg":"uploaded pre-keys"}
saving auth state
{"level":30,"time":1638202395657,"pid":88726,"hostname":"host","msg":"opened connection to WA"}
connection update { connection: 'open' }
{"level":10,"time":1638202396050,"pid":88726,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"edge_routing","attrs":{},"content":[{"tag":"routing_info","attrs":{},"content":{"type":"Buffer","data":[__4 bytes__]}}]}]},"msg":"communication"}
{"level":10,"time":1638202396050,"pid":88726,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"dirty","attrs":{"type":"account_sync","timestamp":"1638202396"},"content":null}]},"msg":"communication"}
saving auth state
{"level":10,"time":1638202396054,"pid":88726,"hostname":"host","msgId":"1.7-1","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"1.7-1"},"content":[{"tag":"active","attrs":{},"content":null}]},"msg":"communication"}
{"level":10,"time":1638202396176,"pid":88726,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"offline","attrs":{"count":"0"},"content":null}]},"msg":"communication"}
{"level":30,"time":1638202396176,"pid":88726,"hostname":"host","msg":"got 0 offline messages/notifications"}
connection update { receivedPendingNotifications: true }
{"level":10,"time":1638202396280,"pid":88726,"hostname":"host","msgId":"1579351387","fromMe":false,"frame":{"tag":"notification","attrs":{"from":"__number__@s.whatsapp.net","type":"account_sync","id":"1579351387","t":"1638202396"},"content":[{"tag":"devices","attrs":{"dhash":"________"},"content":[{"tag":"device","attrs":{"jid":"__number__@s.whatsapp.net"},"content":null},{"tag":"device","attrs":{"jid":"__number__:41@s.whatsapp.net","key-index":"11"},"content":null},{"tag":"device","attrs":{"jid":"__number__:42@s.whatsapp.net"},"content":null},{"tag":"key-index-list","attrs":{"ts":"1638202393"},"content":{"type":"Buffer","data":[________]}}]}]},"msg":"communication"}
{"level":20,"time":1638202396281,"pid":88726,"hostname":"host","recv":{"from":"__number__@s.whatsapp.net","type":"account_sync","id":"1579351387","t":"1638202396"},"sent":{"id":"1579351387","to":"__number__@s.whatsapp.net","class":"notification","type":"account_sync"},"msg":"sent \"notification\" ack"}
{"level":10,"time":1638202396282,"pid":88726,"hostname":"host","msgId":"1.7-3","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"1.7-3"},"content":[{"tag":"privacy","attrs":{},"content":[{"tag":"category","attrs":{"name":"groupadd","value":"all"},"content":null},{"tag":"category","attrs":{"name":"last","value":"none"},"content":null},{"tag":"category","attrs":{"name":"status","value":"all"},"content":null},{"tag":"category","attrs":{"name":"profile","value":"all"},"content":null},{"tag":"category","attrs":{"name":"readreceipts","value":"all"},"content":null}]}]},"msg":"communication"}
{"level":10,"time":1638202396587,"pid":88726,"hostname":"host","msgId":"1.7-2","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"1.7-2"},"content":[{"tag":"list","attrs":{},"content":[{"tag":"item","attrs":{"jid":"__OTHER PHONE NUMBER___@s.whatsapp.net"},"content":null}]}]},"msg":"communication"}
{"level":10,"time":1638202396588,"pid":88726,"hostname":"host","msgId":"1.7-0","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"1.7-0"},"content":null},"msg":"communication"}

(a silence from here)

{"level":10,"time":1638202420279,"pid":88726,"hostname":"host","msgId":"1.7-4","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"1.7-4","t":"1638202420"},"content":null},"msg":"communication"}
{"level":10,"time":1638202420280,"pid":88726,"hostname":"host","msg":"recv keep alive"}

(also silence between the pings here)

{"level":10,"time":1638202445293,"pid":88726,"hostname":"host","msgId":"1.7-5","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"1.7-5","t":"1638202445"},"content":null},"msg":"communication"}
{"level":10,"time":1638202445294,"pid":88726,"hostname":"host","msg":"recv keep alive"}


(I have abort the process here)
baileysNov28 hu$ 
```

related: #936
maybe related: #834

**Comment by HUGEIT** [comment]
@vasani-arpit the scan issue does not happen to me with the browser or desktop apps. 

MD is virtually out of beta at this point. you cannot even leave the """beta""" anymore.

do you get this scan issue on the normal apps? or are you referring to something else with the "timeout"?

**Comment by HUGEIT** [comment]
If I add 

```
if (l1.category == 'peer') {
                    return
```

to `onMessageRecieved` to prevent this message from being acknowledged it will show a similar behavior regarding the "stalling" and being stuck on the loading popup for a minute. 

what's different is that by doing it like this it will at least show the session in the list and not present the "Invalid QR code" popup. 

now I have no idea what else to try.

**Comment by itacirgabral** [comment]
> To Reproduce
> Steps to reproduce the behavior:
> 1. clone the project
> 2. npm i, npm run example
> 3. Scan the QR code, in my case with an iPhone

Usually I'm build it and publish to my local verdaccio npm.
I'm not testing the example. But this little demo is working fine https://github.com/itacirgabral/baileysVanilla

**Comment by HUGEIT** [comment]
@itacirgabral it is very unlikely an issue with the auth creds, the messages that use the creds don't even arrive.

**Comment by jackcokk** [comment]
Coming from #955 that resulted from having such a "ghost session" in my phone after a failed scan attempt.

Since these sessions get transmitted by the server it implies that whenever that happens, *parts* of the login process worked and were registered on the server and then it went stuck from there. I do not know how the pre-key things work but #955 made me think that maybe there are no pre-keys to begin with - however in your log it seems like uploading those was successful so the server should have something.

It may also be possible that it can get stuck on different stages of the login, the question in any case is how we can prevent/handle them in Baileys or on our phones.

**Comment by HUGEIT** [comment]
thank you for reminding me, I also had another ghost scan today. relaunched the session to see if it can receive messages but of course it did not.

the last received message was again the `privacy` one.

I have compared incoming and outgoing messages in Baileys and the website and found a difference if it means something:

in Baileys I cannot find any `abt` requests or the response to that. may this result into WA aborting the request somehow?

![image](https://user-images.githubusercontent.com/94182206/144499775-988c6fbc-62db-4ad6-9bbb-d56433f00c24.png)

the same goes for `xmlns="w" <props />` that the client sends but Baileys does not. 

maybe these can lead to the scan failing?

below you can find my log for today's failed scan.

```
{"level":30,"time":1638439850270,"pid":32893,"hostname":"host","msg":"connected to WA Web"}
{"level":20,"time":1638439850464,"pid":32893,"hostname":"host","msg":"handshake recv from WA Web"}
{"level":30,"time":1638439850472,"pid":32893,"hostname":"host","msg":"handshake complete"}
{"level":30,"time":1638439850472,"pid":32893,"hostname":"host","msg":"not logged in, attempting registration..."}
{"level":10,"time":1638439850786,"pid":32893,"hostname":"host","msgId":"1778774326","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"set","id":"1778774326","xmlns":"md"},"content":[{"tag":"pair-device","attrs":{},"content":[{"tag":"ref","attrs":{},"content":{"type":"Buffer","data":[___]}}]}]},"msg":"communication"}
connection update {
  qr: '2'
}

{"level":10,"time":1638439858973,"pid":32893,"hostname":"host","msgId":"1342752208","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"set","id":"1342752208","xmlns":"md"},"content":[{"tag":"pair-success","attrs":{},"content":[{"tag":"platform","attrs":{"name":"smbi"},"content":null},{"tag":"device-identity","attrs":{},"content":{"type":"Buffer","data":[__]}},{"tag":"biz","attrs":{"name":"hu"},"content":null},{"tag":"device","attrs":{"jid":"__number__:64@s.whatsapp.net"},"content":null}]}]},"msg":"communication"}
{"level":20,"time":1638439858973,"pid":32893,"hostname":"host","msg":"pair success recv"}
{"level":20,"time":1638439859011,"pid":32893,"hostname":"host","msg":"pairing configured successfully"}
{"level":30,"time":1638439859012,"pid":32893,"hostname":"host","jid":"__number__:64@s.whatsapp.net","msg":"registered connection, restart server"}
saving auth state
connection update { isNewLogin: true, qr: undefined }
{"level":30,"time":1638439859015,"pid":32893,"hostname":"host","error":{"data":null,"isBoom":true,"isServer":false,"output":{"statusCode":410,"payload":{"statusCode":410,"error":"Gone","message":"Restart Required"},"headers":{}}},"msg":"connection closed"}
connection update {
  connection: 'close',
  lastDisconnect: {
    error: Error: Restart Required
        at WebSocket.<anonymous> (bbaileys/src/Socket/socket.ts:503:17) {
      data: null,
      isBoom: true,
      isServer: false,
      output: [Object]
    },
    date: 2021-12-02T10:10:59.015Z
  }
}
connection update {
  connection: 'connecting',
  receivedPendingNotifications: false,
  qr: undefined
}
{"level":30,"time":1638439859544,"pid":32893,"hostname":"host","msg":"connected to WA Web"}
{"level":20,"time":1638439859815,"pid":32893,"hostname":"host","msg":"handshake recv from WA Web"}
{"level":30,"time":1638439859838,"pid":32893,"hostname":"host","msg":"handshake complete"}
{"level":30,"time":1638439859839,"pid":32893,"hostname":"host","msg":"logging in..."}
{"level":10,"time":1638439860338,"pid":32893,"hostname":"host","fromMe":false,"frame":{"tag":"success","attrs":{"t":"1638439860","props":"26","location":"atn","abprops":"0","creation":"1638439858","companion_enc_static":"__"},"content":null},"msg":"communication"}
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
saving auth state
{"level":30,"time":1638439860394,"pid":32893,"hostname":"host","msg":"uploaded pre-keys"}
saving auth state
{"level":30,"time":1638439860397,"pid":32893,"hostname":"host","msg":"opened connection to WA"}
connection update { connection: 'open' }
{"level":10,"time":1638439860428,"pid":32893,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"edge_routing","attrs":{},"content":[{"tag":"routing_info","attrs":{},"content":{"type":"Buffer","data":____]}}]}]},"msg":"communication"}
{"level":10,"time":1638439860428,"pid":32893,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"dirty","attrs":{"type":"account_sync","timestamp":"1638439860"},"content":null}]},"msg":"communication"}
saving auth state
{"level":10,"time":1638439860578,"pid":32893,"hostname":"host","msgId":"2.d-1","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"2.d-1"},"content":[{"tag":"active","attrs":{},"content":null}]},"msg":"communication"}
{"level":10,"time":1638439860619,"pid":32893,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"offline","attrs":{"count":"0"},"content":null}]},"msg":"communication"}
{"level":30,"time":1638439860619,"pid":32893,"hostname":"host","msg":"got 0 offline messages/notifications"}
connection update { receivedPendingNotifications: true }
{"level":10,"time":1638439860643,"pid":32893,"hostname":"host","msgId":"3417445962","fromMe":false,"frame":{"tag":"notification","attrs":{"from":"__number__@s.whatsapp.net","type":"account_sync","id":"3417445962","t":"1638439860"},"content":[{"tag":"devices","attrs":{"dhash":"2:__"},"content":[{"tag":"device","attrs":{"jid":"__number__@s.whatsapp.net"},"content":null},{"tag":"device","attrs":{"jid":"__number__:48@s.whatsapp.net","key-index":"3"},"content":null},{"tag":"device","attrs":{"jid":"__number__:63@s.whatsapp.net","key-index":"28"},"content":null},{"tag":"device","attrs":{"jid":"__number__:64@s.whatsapp.net"},"content":null},{"tag":"key-index-list","attrs":{"ts":"1638439856"},"content":{"type":"Buffer","data":[___]}}]}]},"msg":"communication"}
{"level":20,"time":1638439860643,"pid":32893,"hostname":"host","recv":{"from":"__number__@s.whatsapp.net","type":"account_sync","id":"3417445962","t":"1638439860"},"sent":{"id":"3417445962","to":"__number__@s.whatsapp.net","class":"notification","type":"account_sync"},"msg":"sent \"notification\" ack"}
{"level":10,"time":1638439860645,"pid":32893,"hostname":"host","msgId":"2.d-2","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"2.d-2"},"content":[{"tag":"list","attrs":{},"content":null}]},"msg":"communication"}
{"level":10,"time":1638439860649,"pid":32893,"hostname":"host","msgId":"2.d-3","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"2.d-3"},"content":[{"tag":"privacy","attrs":{},"content":[{"tag":"category","attrs":{"name":"groupadd","value":"all"},"content":null},{"tag":"category","attrs":{"name":"last","value":"all"},"content":null},{"tag":"category","attrs":{"name":"status","value":"all"},"content":null},{"tag":"category","attrs":{"name":"profile","value":"all"},"content":null},{"tag":"category","attrs":{"name":"readreceipts","value":"all"},"content":null}]}]},"msg":"communication"}
{"level":10,"time":1638439860719,"pid":32893,"hostname":"host","msgId":"2.d-0","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"2.d-0"},"content":null},"msg":"communication"}
{"level":10,"time":1638439885013,"pid":32893,"hostname":"host","msgId":"2.d-4","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"2.d-4","t":"1638439884"},"content":null},"msg":"communication"}
{"level":10,"time":1638439885013,"pid":32893,"hostname":"host","msg":"recv keep alive"}
{"level":10,"time":1638439910019,"pid":32893,"hostname":"host","msgId":"2.d-5","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"2.d-5","t":"1638439910"},"content":null},"msg":"communication"}
{"level":10,"time":1638439910020,"pid":32893,"hostname":"host","msg":"recv keep alive"}

(nothing more happening)
```


```
(restarted session to see what happens when I send a message)

connection update {
  connection: 'connecting',
  receivedPendingNotifications: false,
  qr: undefined
}
{"level":30,"time":1638439962605,"pid":32907,"hostname":"host","msg":"connected to WA Web"}
{"level":20,"time":1638439962646,"pid":32907,"hostname":"host","msg":"handshake recv from WA Web"}
{"level":30,"time":1638439962653,"pid":32907,"hostname":"host","msg":"handshake complete"}
{"level":30,"time":1638439962653,"pid":32907,"hostname":"host","msg":"logging in..."}
{"level":10,"time":1638439963008,"pid":32907,"hostname":"host","fromMe":false,"frame":{"tag":"success","attrs":{"t":"1638439963","props":"26","location":"odn","abprops":"0","creation":"1638439858","companion_enc_static":"_____"},"content":null},"msg":"communication"}
{"level":30,"time":1638439963010,"pid":32907,"hostname":"host","msg":"opened connection to WA"}
connection update { connection: 'open' }
{"level":10,"time":1638439963198,"pid":32907,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"edge_routing","attrs":{},"content":[{"tag":"routing_info","attrs":{},"content":{"type":"Buffer","data":[__]}}]}]},"msg":"communication"}
{"level":10,"time":1638439963199,"pid":32907,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"dirty","attrs":{"type":"account_sync","timestamp":"1638439963"},"content":null}]},"msg":"communication"}
{"level":30,"time":1638439963200,"pid":32907,"hostname":"host","fromTimestamp":1638439860,"msg":"requesting account sync"}
{"level":10,"time":1638439963201,"pid":32907,"hostname":"host","msgId":"e.c-0","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"e.c-0"},"content":[{"tag":"active","attrs":{},"content":null}]},"msg":"communication"}
saving auth state
{"level":10,"time":1638439963236,"pid":32907,"hostname":"host","msgId":"e.c-3","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"e.c-3"},"content":null},"msg":"communication"}
{"level":10,"time":1638439963281,"pid":32907,"hostname":"host","fromMe":false,"frame":{"tag":"ib","attrs":{"from":"@s.whatsapp.net"},"content":[{"tag":"offline","attrs":{"count":"0"},"content":null}]},"msg":"communication"}
{"level":30,"time":1638439963282,"pid":32907,"hostname":"host","msg":"got 0 offline messages/notifications"}
connection update { receivedPendingNotifications: true }
{"level":10,"time":1638439963338,"pid":32907,"hostname":"host","msgId":"e.c-2","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"e.c-2"},"content":[{"tag":"privacy","attrs":{},"content":[{"tag":"category","attrs":{"name":"groupadd","value":"all"},"content":null},{"tag":"category","attrs":{"name":"last","value":"all"},"content":null},{"tag":"category","attrs":{"name":"status","value":"all"},"content":null},{"tag":"category","attrs":{"name":"profile","value":"all"},"content":null},{"tag":"category","attrs":{"name":"readreceipts","value":"all"},"content":null}]}]},"msg":"communication"}
{"level":10,"time":1638439963360,"pid":32907,"hostname":"host","msgId":"e.c-1","fromMe":false,"frame":{"tag":"iq","attrs":{"from":"@s.whatsapp.net","type":"result","id":"e.c-1"},"content":[{"tag":"list","attrs":{},"content":null}]},"msg":"communication"}
{"level":10,"time":1638439972500,"pid":32907,"hostname":"host","msgId":"3A20290DC762C0ECB445","fromMe":false,"frame":{"tag":"message","attrs":{"from":"__number__-123456@g.us","type":"text","id":"3A20290DC762C0ECB445","participant":"__number__@s.whatsapp.net","t":"1638439972","count":"2","verified_level":"unknown","notify":"hu","verified_name":"___"},"content":[{"tag":"verified_name","attrs":{"v":"1","verified_level":"unknown"},"content":{"type":"Buffer","data":[____]}}]},"msg":"communication"}
{"level":50,"time":1638439972513,"pid":32907,"hostname":"host","msgId":"3A20290DC762C0ECB445","trace":"Error: No sender key for: __number__-123456@g.us::__number__::0\n    at GroupCipher.decrypt (bbaileys/WASignalGroup/group_cipher.js:39:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.decodeMessageStanza (bbaileys/src/Utils/decode-wa-message.ts:78:37)\n    at async WebSocket.<anonymous> (bbaileys/src/Socket/messages-recv.ts:348:21)","data":"___","msg":"failure in decrypting message"}
saving auth state
{"level":30,"time":1638439972530,"pid":32907,"hostname":"host","msgAttrs":{"from":"__number__-123456@g.us","type":"text","id":"3A20290DC762C0ECB445","participant":"__number__@s.whatsapp.net","t":"1638439972","count":"2","verified_level":"unknown","notify":"hu","verified_name":"39130791736396876"},"retryCount":1,"msg":"sent retry receipt"}
saving auth state
saving auth state
messages.upsert
contacts.update
[{"id":"__number__@s.whatsapp.net","notify":"hu"}]
{"level":10,"time":1638439972567,"pid":32907,"hostname":"host","msgId":"3A20290DC762C0ECB445","fromMe":false,"frame":{"tag":"ack","attrs":{"from":"__number__-123456@g.us","class":"receipt","type":"retry","id":"3A20290DC762C0ECB445","participant":"__number__@s.whatsapp.net"},"content":null},"msg":"communication"}
{"level":10,"time":1638439974028,"pid":32907,"hostname":"host","msgId":"3A20290DC762C0ECB445","fromMe":false,"frame":{"tag":"receipt","attrs":{"from":"__number__-123456@g.us","id":"3A20290DC762C0ECB445","participant":"__other number__@s.whatsapp.net","t":"1638439974"},"content":null},"msg":"communication"}
messages.update
{"level":20,"time":1638439974030,"pid":32907,"hostname":"host","recv":{"from":"__number__-123456@g.us","id":"3A20290DC762C0ECB445","participant":"__other number__@s.whatsapp.net","t":"1638439974"},"sent":{"id":"3A20290DC762C0ECB445","to":"__number__-123456@g.us","class":"receipt","participant":"__other number__@s.whatsapp.net"},"msg":"sent \"receipt\" ack"}
{"level":10,"time":1638439974379,"pid":32907,"hostname":"host","msgId":"3A20290DC762C0ECB445","fromMe":false,"frame":{"tag":"receipt","attrs":{"from":"__number__-123456@g.us","id":"3A20290DC762C0ECB445","participant":"__another number__@s.whatsapp.net","t":"1638439974"},"content":null},"msg":"communication"}
messages.update
{"level":20,"time":1638439974380,"pid":32907,"hostname":"host","recv":{"from":"__number__-123456@g.us","id":"3A20290DC762C0ECB445","participant":"__another number__@s.whatsapp.net","t":"1638439974"},"sent":{"id":"3A20290DC762C0ECB445","to":"__number__-123456@g.us","class":"receipt","participant":"_another number__@s.whatsapp.net"},"msg":"sent \"receipt\" ack"}

```

**Comment by HUGEIT** [comment]
I am able to send both of these events, they both return data from the socket that looks like the original thing.

```
    const sendAbt = (tag: 'props') => (
        sendNode({
            tag: 'iq',
            attrs: {
                to: S_WHATSAPP_NET,
                xmlns: 'abt',
                type: 'get',
                id: generateMessageTag(),
            },
            content: [
                { tag, attrs: { 
                    protocol: "1"
                    } 
                }
            ]
        })
    )

    const sendProps = (tag: 'props') => (
        sendNode({
            tag: 'iq',
            attrs: {
                to: S_WHATSAPP_NET,
                xmlns: 'w',
                type: 'get',
                id: generateMessageTag(),
            },
            content: [
                { tag, attrs: { } }
            ]
        })
    )
```

we could probably make this a bit smarter but the definition somehow forces me to use the function parameter because it doesn't allow setting a string directly for some reason. the question is if there are any events missing, if that solves the issue and @adiwajshing shouldn't we send these anyway? I think, even in the case of this not helping for the QR code we would still be better off having that included in Baileys to stay as close to the original as possible.

**Comment by adiwajshing** [comment]
@HUGEIT I'm not sure how updating the logout function fixes the QR issue -- is there a connection there?

**Comment by jackcokk** [comment]
@adiwajshing He did not change the logout function. It is the function for successful login.

![image](https://user-images.githubusercontent.com/91759091/144577259-bd0df21b-2fd1-4887-b8f8-bc05abeb041d.png)

Maybe GitHub does some bad regex. If you view by file you will see the raw location: https://github.com/HUGEIT/Baileys/blob/e2be8dece56e2db73279d1e1999a352a93aba8b9/src/Socket/socket.ts#L531 

One issue I can see with that is that it will poll it each time it connects instead of only after sign-ups. I can only find one of these in the websocket traffic on a reconnect after successful signup but it may also be due to me not being able to inject the websocket intercept quickly enough.

**Comment by adiwajshing** [comment]
yeah you're right -- the diff was a little weird, I didn't read it correctly -- @HUGEIT can open a PR for this pls

**Comment by HUGEIT** [comment]
yes #959 
![Alt Text](https://c.tenor.com/YgU-B2bf_rEAAAAC/ai-sir-happy.gif)

but please check the commit message - I am unsure about whether we need `await` there or if it simply slows down unnecessarily

**Comment by jackcokk** [comment]
I tried 2 scans for now and they worked with your PR @HUGEIT 

It is hard to tell at this point in time since not every scan fails but if that makes it work we could finally use it in production.

@adiwajshing have you considered something for donations like a sponsor button? I think as soon as the QR is resolved, Baileys-md is complete to be merged into master. Unless I am mistaken the only currently-open issue about unexpected behavior that contains a stack trace is #928
